Sources/SwifterSwift/UIKit/UIButtonExtensions.swift

```diff
@@ -167,34 +167,72 @@ public extension UIButton {
         states.forEach { setTitle(title, for: $0) }
     }
 
-    /// SwifterSwift: Center align title text and image
+    /// SwifterSwift: Set the image to the left of the text
+    ///
     /// - Parameters:
-    ///   - imageAboveText: set true to make image above title text, default is false, image on left of text
     ///   - spacing: spacing between title text and image.
-    func centerTextAndImage(imageAboveText: Bool = false, spacing: CGFloat) {
-        if imageAboveText {
-            // https://stackoverflow.com/questions/2451223/#7199529
-            guard
-                let imageSize = imageView?.image?.size,
-                let text = titleLabel?.text,
-                let font = titleLabel?.font else { return }
-
-            let titleSize = text.size(withAttributes: [.font: font])
-
-            let titleOffset = -(imageSize.height + spacing)
-            titleEdgeInsets = UIEdgeInsets(top: 0.0, left: -imageSize.width, bottom: titleOffset, right: 0.0)
-
-            let imageOffset = -(titleSize.height + spacing)
-            imageEdgeInsets = UIEdgeInsets(top: imageOffset, left: 0.0, bottom: 0.0, right: -titleSize.width)
-
-            let edgeOffset = abs(titleSize.height - imageSize.height) / 2.0
-            contentEdgeInsets = UIEdgeInsets(top: edgeOffset, left: 0.0, bottom: edgeOffset, right: 0.0)
-        } else {
-            let insetAmount = spacing / 2
-            imageEdgeInsets = UIEdgeInsets(top: 0, left: -insetAmount, bottom: 0, right: insetAmount)
-            titleEdgeInsets = UIEdgeInsets(top: 0, left: insetAmount, bottom: 0, right: -insetAmount)
-            contentEdgeInsets = UIEdgeInsets(top: 0, left: insetAmount, bottom: 0, right: insetAmount)
-        }
+    func setImageLeftOfText(spacing: CGFloat) {
+        let insetAmount = spacing / 2
+        imageEdgeInsets = UIEdgeInsets(top: 0, left: -insetAmount, bottom: 0, right: insetAmount)
+        titleEdgeInsets = UIEdgeInsets(top: 0, left: insetAmount, bottom: 0, right: -insetAmount)
+        contentEdgeInsets = UIEdgeInsets(top: 0, left: insetAmount, bottom: 0, right: insetAmount)
+    }
+
+    /// SwifterSwift: Set the image to the right of the text
+    ///
+    /// - Parameters:
+    ///   - spacing: spacing between title text and image.
+    func setImageRightOfText(spacing: CGFloat) {
+        semanticContentAttribute = .forceRightToLeft
```

Comment:
No, please don't do this hack.

Reply:
As I said above, if you don’t use this method, no matter how you set contentEdgeInsets, when the spacing is too large, the image will always exceed the bounds of the button. Unless we don’t consider whether it exceeds the bounds of the button....

Reply:
I can see the problem now. You're right, when the button is too small to fit the content then everything breaks. This is why I would never actually use any of these methods because it breaks so easily, either in these scenarios (where the content is too large) or if the view ever needs to be laid out again. I'll keep looking into it 😉

Reply:
Yes, but this can meet most of the needs, but it is not perfect.